src/Integration/MefServices/ActiveSolutionBoundTracker.cs

```diff
@@ -97,6 +132,15 @@ private void RaiseAnalyzersChangedIfBindingChanged()
             }
         }
 
+        #region IPartImportsSatisfiedNotification
+
+        public void OnImportsSatisfied()
+        {
+            UpdateConnection();
```

Comment:
This method is called once MEF composition has completed successfully  i.e. after all of the objects have been constructed and wired together.
It's safer to not attempt the connection until this point in case it triggers any side effects that fail if all of the MEF objects have not been created.